bundledata.go

```diff
@@ -291,13 +309,30 @@ func (bd *BundleData) normalizeData() error {
 		if app == nil {
 			continue
 		}
+		if app.Empty() {
+			app.Charm = appName
+			app.NumUnits = newInt(1)
```

Comment:
subordinates can't be deployed with a count.

Reply:
So there is indeed a problem. Essentially we don't have enough information at the time of unpacking to determine if a charm is a subordinate. As we only have a charm URL and they do not identify a subordinate and only the charm metadata expresses this option, we are stuck.

The way I implemented it was that subordinates will never be allowed a short syntax, there isn't anything we can do about it with the current setup.

As this package is a modeling library where we don't actually confirm the existence of a given charm, just values appear to be correct for deployment, we're in a bit of a sticky situation.

----

The previous approach was to use `VerifyWithCharms` which requires that you know all the charms upfront. This isn't the case at all with Juju. So that was a fallacy.

----

In reality, we probably want to expose a `NumUnits` method that we use to get the correct number of default units and use that throughout Juju. We can check the `alteredApplications` method to see if the subordinate was using the short syntax originally.

Reply:
I don't believe Juju itself uses NumUnits from the bundle code directly and actually we might only be using from bundlechanges, we could use the `NumUnits(subordinate)` to gain that information.